recipes-core/ssh-mode-switch/ssh-mode-switch.inc

```diff
@@ -0,0 +1,65 @@
+# Generic SSH user setup include file
+# Required variables:
+# - SSH_USER: Username to create
+# - SSH_USER_ID: Numeric user ID
+# - SSH_USER_KEY: SSH public key for the user
+# Optional variables:
+# - SSH_USER_SHELL: Shell for the user (defaults to /bin/sh)
+# - SSH_USER_HOME: Home directory (defaults to /home/${SSH_USER})
+
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+inherit useradd
+
+SRC_URI = "file://${BPN}.sh"
+
+# Set defaults if not defined
+SSH_USER ?= "${PN}"
+SSH_USER_KEY ?= "${SEARCHER_SSH_KEY}"
+SSH_USER_SHELL ?= "/bin/${PN}"
+SSH_USER_HOME ?= "/home/${SSH_USER}"
+
+USERADD_PACKAGES = "${PN}"
+USERADD_PARAM:${PN} = "-m -d ${SSH_USER_HOME} -s ${SSH_USER_SHELL} -u ${SSH_USER_ID} ${SSH_USER}"
+
+python () {
+    # Check if SSH_USER_KEY is set in the environment or in local.conf
+    ssh_key = d.getVar('SSH_USER_KEY')
+    
+    if ssh_key is None:
+        # If not set, check the original environment
+        origenv = d.getVar("BB_ORIGENV", False)
+        if origenv:
+            ssh_key = origenv.getVar('SSH_USER_KEY')
```

Comment:
can we use the already existing env var SEARCHER_SSH_KEY instead of introducing a new one?